### DISCUSSION TITLE
PiVPN on docker

### DISCUSSION THREAD

**Discussion by joaopinto14** [pivpn/pivpn#1500] [Feature requests]
Installing PiVPN with docker would be cleaner and safer.
With association with Pihole (Docker)

**Comment by ghost** [comment]
Yeah, docker works perfectly. But two minor limiations: 
1. Docker doesn't like bridge interfaces if it's not under Linux env
2. Docker is not mainstream with small customers

**Reply by ghost** [reply]
I used angristan script, which worked for my needs, it has a clean headless install and uninstall options: https://github.com/angristan/openvpn-install Minor script modifications allow doing this flawlessly. PiVPN is also nice, but I guess it's designed for stable environments and not for temporary deployments. I tried playing with docker, not something I need now. I prefere VMs under VirtualBox, VMWare or HyperV systems which fully supports bridged networking so I have full access from the VM to the system network. OpenVPN in docker is a good idea, but only for locally deployed dockers within the docker network.

**Comment by coolapso** [comment]
your comment tells me clearly you just discovered docker and you don't really know what you are asking. 

1 - Iptables & Networking does not work the same way inside docker containers
2 - you need to account for the certificates generation and make sure that not everybody ends up having the same certificates.  And this is the reason i Strongly discourage anyone using the available Docker images around! at least not without reviewing the source code properly. 
3 - It forces a MAJOR refactor on the whole pivpn code. 

So now that you have the guidelines ... I am more than glad to review a pull request that accomplishes what I have (so far) failed to do! 

Moving to discussions.
